src/librustdoc/clean/types.rs

```diff
@@ -1375,55 +1375,6 @@ crate struct PolyTrait {
     crate generic_params: Vec<GenericParamDef>,
 }
 
-/// A representation of a type suitable for hyperlinking purposes. Ideally, one can get the original
-/// type out of the AST/`TyCtxt` given one of these, if more information is needed. Most
-/// importantly, it does not preserve mutability or boxes.
-#[derive(Clone, PartialEq, Eq, Debug, Hash)]
-crate enum Type {
```

Comment:
Moving `Type` and its impls will cause a **lot** of conflicts with my "Cleanup `clean`" PR and probably other PRs as well. Can you please revert this part of the change? I'm okay with keeping the merging of the two inherent impls on `Type` (the general one and the `inner_def_id` one), and maybe also moving `trait GetDefId { ... }` so it's not in the middle of the `Type` code, but most of the other changes feel like unnecessary churn to me.

Reply:
Ok!